### ISSUE TITLE
Set the secure flag in MOD_AUTH_CAS cookie

### ISSUE THREAD

**Issue by hshyk** [apereo/mod_auth_cas#139]
Hi,

Is it possible to send down the Secure flag with the MOD_AUTH_CAS cookie? I found a directive CASCookieHttpOnly which sets the HttpOnly flag, but not one for the Secure flag.  I tried setting it via Apache configuration `Header always edit Set-Cookie ^(.*)$ $1;HttpOnly;Secure`. It works for application cookies, but does not for the CAS cookie.

**Comment by pames** [comment]
mod_auth_cas uses 2 cookies, one for secure channels (https), and one for insecure channels (http). 
 MOD_AUTH_CAS is the http cookie, MOD_AUTH_CAS_S is the https cookie (https://github.com/apereo/mod_auth_cas/blob/master/src/mod_auth_cas.h#L94)

The MOD_AUTH_CAS cookie is ignored over https:// connections :
https://github.com/apereo/mod_auth_cas/blob/master/src/mod_auth_cas.c#L2081

The logic for setting the secure flag is controlled here:
https://github.com/apereo/mod_auth_cas/blob/master/src/mod_auth_cas.c#L758

And whether or not the connection is considered https is determined by this function:
https://github.com/apereo/mod_auth_cas/blob/master/src/mod_auth_cas.c#L424

So in short, if you are connecting to the CAS server over https://, it should automatically set that flag.

**Comment by hshyk** [comment]
I see the issue now. If you offload the SSL via a proxy, the Apache server will serve the HTTP cookie. I set Apache to use SSL as well and now it is working. Thanks!
